### PR TITLE
8306072: Open source several AWT MouseInfo related tests

### PR DESCRIPTION
**Pull request by TejeshR13**
This open source AWT Mouse Info related tests.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8306072](https://bugs.openjdk.org/browse/JDK-8306072): Open source several AWT MouseInfo related tests

### Reviewers
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**) ⚠️ Review applies to [86ba000c](https://git.openjdk.org/jdk/pull/13573/files/86ba000ced1befcc2a404e1ffa5e07b79d3ba522)
 * [Prasanta Sadhukhan](https://openjdk.org/census#psadhukhan) (@prsadhuk - **Reviewer**) ⚠️ Review applies to [86ba000c](https://git.openjdk.org/jdk/pull/13573/files/86ba000ced1befcc2a404e1ffa5e07b79d3ba522)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13573/head:pull/13573` \
`$ git checkout pull/13573`

Update a local copy of the PR: \
`$ git checkout pull/13573` \
`$ git pull https://git.openjdk.org/jdk.git pull/13573/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13573`

View PR using the GUI difftool: \
`$ git pr show -t 13573`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13573.diff">https://git.openjdk.org/jdk/pull/13573.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13573#issuecomment-1517280421)

### CODE REVIEW

File: test/jdk/java/awt/MouseInfo/ContainerMousePositionTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            if (pMousePosition != null) {
+                throw new RuntimeException("Test failed: Pointer was outside of " + "the Frame widow and getMousePosition() should return null");
+            }
+            System.out.println("Test stage completed: Pointer was outside of " + "the Frame widow and getMousePosition() returned null");
+
+            robot.delay(DELAY);
+
+            System.out.println("ComponentMousePositionTest PASSED.");
+        } finally {
+            if (frame != null) {
```
Comment (mrserb) [inline test/jdk/java/awt/MouseInfo/ContainerMousePositionTest.java:136]: frame var is assigned on EDT and then used on non-EDT thread.

File: test/jdk/java/awt/MouseInfo/ContainerMousePositionTest.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            Point p = button.getLocationOnScreen();
+            robot.mouseMove(p.x + button.getWidth() / 2, p.y + button.getHeight() / 2);
+
+            robot.delay(DELAY);
+
+            Point pMousePosition = frame.getMousePosition(false);
+            if (pMousePosition != null) {
+                throw new RuntimeException("Test failed: Container is overlapped by " + " child and it should be taken into account");
+            }
+            System.out.println("Test stage completed: Container is overlapped by " + " child and it was taken into account");
```
Comment (lawrence-andrew) [inline test/jdk/java/awt/MouseInfo/ContainerMousePositionTest.java:92]: Can you limit to 80 column length.

File: test/jdk/java/awt/MouseInfo/ContainerMousePositionTest.java (hunk 1 above)
Comment (TejeshR13) [reply test/jdk/java/awt/MouseInfo/ContainerMousePositionTest.java]: Updated.

File: test/jdk/java/awt/MouseInfo/ContainerMousePositionTest.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    }
+
+    public void start() throws Exception {
+        try {
+            robot.delay(DELAY);
+            robot.waitForIdle();
+
+            EventQueue.invokeAndWait(() -> {
+                Point p = button.getLocationOnScreen();
+                xPos = p.x + button.getWidth() / 2;
```
Comment (lawrence-andrew) [inline test/jdk/java/awt/MouseInfo/ContainerMousePositionTest.java:88]: xPos & yPos are used within EDT and outside EDT ?

Comment (TejeshR13) [reply test/jdk/java/awt/MouseInfo/ContainerMousePositionTest.java:88]: Yeah, getting the values within EDT and used for `Robot` operations outside. Isn't it allowed ?

Comment (lawrence-andrew) [reply test/jdk/java/awt/MouseInfo/ContainerMousePositionTest.java:88]: It is allowed please use transient keyword for xPos & yPos

Comment (NikitkoCent) [reply test/jdk/java/awt/MouseInfo/ContainerMousePositionTest.java:88]: @lawrence-andrew, BTW, why `transient`? Should it be `volatile` instead?

**Review by mrserb — APPROVED** [review]
Looks fine, please confirm that mach5 is green.

**Comment by TejeshR13** [comment]
> Looks fine, please confirm that mach5 is green.

Yes, I retested with mach5 and its fine.

File: test/jdk/java/awt/MouseInfo/ContainerMousePositionTest.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+import java.awt.Button;
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Panel;
+import java.awt.Point;
+import java.awt.Robot;
+
+import java.lang.reflect.InvocationTargetException;
```
Comment (prsadhuk) [inline test/jdk/java/awt/MouseInfo/ContainerMousePositionTest.java:39]: guess it's not needed now..

**Review by prsadhuk — APPROVED** [review]

File: test/jdk/java/awt/MouseInfo/ContainerMousePositionTest.java (hunk 4 above)
Comment (TejeshR13) [reply test/jdk/java/awt/MouseInfo/ContainerMousePositionTest.java]: Updated.

**Pull request closed without merging**